modules/csg/csg.cpp

```diff
@@ -242,7 +242,7 @@ void CSGBrushOperation::BuildPoly::_clip_segment(const CSGBrush *p_brush, int p_
 	//check if edge and poly share a vertex, of so, assign it to segment_idx
 	for (int i = 0; i < points.size(); i++) {
 		for (int j = 0; j < 2; j++) {
-			if (Math::is_zero_approx(segment[j].distance_to(points[i].point))) {
+			if (segment[j].distance_to(points[i].point) < 0.00005) {
```

Comment:
There should probably be a comment that justifies the magic number's presence :slightly_smiling_face:

Alternatively, could `CMP_EPSILON * 2` be used here?

Reply:
Not sure if 0.00005 is large enough. It's enough to fix the example in the issue, but maybe there are other cases where it isn't.

Reply:
Can you investigate is_equal_approx_ratio?